gossipd/routing.c

```diff
@@ -333,28 +331,22 @@ static u64 risk_fee(u64 amount, u32 delay, double riskfactor)
 /* We track totals, rather than costs.  That's because the fee depends
  * on the current amount passing through. */
 static void bfg_one_edge(struct node *node, size_t edgenum, double riskfactor,
-			 double fuzz, const struct isaac64_ctx *baserng)
+			 double fuzz, const struct siphash_seed *base_seed)
 {
 	struct node_connection *c = node->in[edgenum];
 	size_t h;
-	struct isaac64_ctx myrng;
 	double fee_scale = 1.0;
-	u8 *scid;
 
 	if (fuzz != 0.0) {
-		/* Copy RNG state */
-		myrng = *baserng;
-
-		/* Provide the channel short ID, as additional
-		 * entropy. */
-		scid = tal_arr(NULL, u8, 0);
-		towire_short_channel_id(&scid, &c->short_channel_id);
-		isaac64_reseed(&myrng,
-			       (const unsigned char *) scid, tal_len(scid));
-		tal_free(scid);
+		u64 scid = short_channel_id_to_uint(&c->short_channel_id);
+		u64 h =	siphash24(base_seed, &scid, sizeof(scid));
 
 		/* Scale fees for this channel */
-		fee_scale = 1.0 + fuzz * isaac64_next_signed_double(&myrng);
+		/* rand = (h / UINT64_MAX)  random number between 0.0 -> 1.0
+		 * 2*fuzz*rand              random number between 0.0 -> 2*fuzz
+		 * 2*fuzz*rand - fuzz       random number between -fuzz -> +fuzz
+		 */
+		fee_scale = 1.0 + (2.0 * fuzz * h / UINT64_MAX) - fuzz;
```

Comment:
Pretty sure we can just make h an s64 and divide by INT64_MAX: `fee_scale = 1.0 + fuzz * h / INT64_MAX`

But it's trivial and unlikely to be any faster, so I'll just leave this as a comment.

Reply:
Negative number absolute range is one higher than positive range, so getting INT64_MIN by chance would break that. Of course very unlikely but well.